DashAI/front/src/components/ConfigurableObject/FormInputs.jsx

```diff
@@ -0,0 +1,377 @@
+import React, { useState, useEffect } from "react";
+import {
+  TextField,
+  MenuItem,
+  Stack,
+  Typography,
+  IconButton,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+  FormControlLabel,
+  Checkbox,
+  FormControl,
+} from "@mui/material";
+import SettingsIcon from "@mui/icons-material/Settings";
+import { generateTooltip } from "../ParameterForm";
+import PropTypes from "prop-types";
+import { getDefaultValues } from "../../utils/values";
+import Subform from "./Subform";
+import { styled } from "@mui/material/styles";
+
+const Input = styled(TextField)(({ theme }) => ({
+  minWidth: "20vw",
+}));
+
+export function Integer({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        type="number"
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Integer.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Integer.defaultProps = {
+  error: undefined,
+};
+
+// This field is designed to handle non-integer numeric inputs
+export function Number({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Number.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Number.defaultProps = {
+  error: undefined,
+};
+
+export function Float({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        type="number"
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Float.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Float.defaultProps = {
+  error: undefined,
+};
+
+export function Text({ name, value, onChange, error, description }) {
```

Comment:
Move to `front/src/components/ConfigurableObject/Inputs/TextInput.jsx`

```suggestion
export function TextInput({ name, value, onChange, error, description }) {
```